### ISSUE TITLE
Excluding bytes=0-1 requests

### ISSUE THREAD

**Issue by jdelStrother** [open-downloads/odl#3]
The readme says...

> It can't be a streaming request for the first 2 bytes (i.e. Range: bytes=0-1)

Doesn't this discount a lot of genuine Apple Podcast plays, given a certain server setup?

Say your podcast host receives a request for example.com/foo.mp3, then responds with a 302 redirect to a second server that actually hosts the file (eg a CDN).

If you try to stream an mp3 in Apple Podcasts (at least on iOS 12.4), the podcasts app sends a single `Range: bytes=0-1` to example.com, then follows the 302 redirect to the CDN.  Any subsequent requests are sent directly to the CDN.
In order to count such plays, you'd need access to the CDN logs. Is that expected? It would also seem to prevent tracking-prefix services (Chartable et al) from being able to provide accurate oDL numbers.

**Comment by johnloringpollard** [comment]
Yea but somehow Podtrac is IAB certified even though IAB says you need to check and ensure 60s of audio has been downloaded. So I'm not sure how they got approved.

I wonder if there needs to be 2 different standards one reading CDN logs and the other reading the http requests before being sent to a CDN.  The former would be for hosts and the latter would be for podtrac, chartable etc.

**Comment by screeley** [comment]
In theory, yes, but in practice, no. 

We have compared our oDL Analytics Prefix numbers to server logs from an IAB certified host with a 302 redirect, and there is under a 1% difference. 

The vast majority of these 0-1 requests are just to ensure that the episode can be streamed. We will generally see more requests at 32,768 and 65,536, which is not a minute of audio, but we know at least the stream is started, so it's counted. Sometimes, we won't see the 32 or 64, but it will randomly request from much later in the episode.

@dzohrob has a theory around keep-alives that I like. 

The numbers will be slightly different from Prefix to Server logs, but not meaningfully enough to split the spec into two.

**Comment by jdelStrother** [comment]
> @dzohrob has a theory around keep-alives that I like

_Interesting_. I’ll investigate some more

**Comment by pbaveja** [comment]
Late to the conversation but any hint about the keep-alive theory to somewhat accurately track solely with a redirect request log?

**Comment by jdelStrother** [comment]
No, sorry. We ended up switching to parsing CDN logs before I got any real testing done.

**Comment by aelitneg** [comment]
I know this is an old conversation, but I wanted to share some current research I've done working in this space for the past 8 months. I want to preface this that I am a software engineer with limited domain knowledge of the podcast space. I can only share what I have seen building out our hosting platform. 

Most of my testing was done in Jan / Feb 2023. I no longer see Apple Podcasts requesting the `bytes=0-1` requests, however I am now seeing this on all Safari based browsers / web apps. We also use a HTTP 302 redirect to a CDN. Safari repeats the whole bytes=0-1 sequence again with the redirected URL. In our instance, it really looks like we would be more "accurate" if we _did not_ filter these requests out as we log requests to the original URL. 

It would be good to get some additional guidance on this topic 😄
